utils/filter.c

```diff
@@ -780,13 +780,34 @@ static int add_trigger_entry(struct rb_root *root,
 	return ret;
 }
 
+static char* convert_loc(char* patt)
+{
+	char *ret;
+
+	/* remove trailing '/' before conversion to regex */
+	if (patt[strlen(patt) - 1] == '/')
+		patt[strlen(patt) - 1] = '\0';
+
+	/*
+	 * convert to a regex string to match all instances of including patt
+	 * in the path. exceptionally, if patt starts at the root, convert only
+	 * the right.
+	 * for example, if patt is 'cmds', it matches 'cmds', 'cmds/', 'a/cmds'
+	 * and 'a/cmds/b'. it matches 'abc.c', 'a/abc.c' if patt is 'abc.c'.
+	 */
+	xasprintf(&ret, "%s%s%s", (patt[0] == '/')? "" : "(^|(.*/))",
+		  patt, "($|(/.*))");
```

Comment:
We should support `--match=glob` case as well.  Also, shouldn't it start with `^` when it's a absolute pathname?

Reply:
> when it's a absolute pathname?

To help understand, could you please give me an example of patt and loc for this case?

Reply:
Sorry for the late reply.  For glob matching, "cmds" should be converted to "cmds/*" (without the ".") and I was thinking about "/usr/bin/true" might match with "/opt/sysroot/usr/bin/true".